### ISSUE TITLE
Can't make the toast message uses the whole screen width

### ISSUE THREAD

**Issue by sdumjahn** [calintamas/react-native-toast-message#282]
**Describe the bug**
In previous versions I could set the width of the toast message to 100% and the messages uses the whole screen width. 
Since the version 2 this does not work anymore.

**Steps to reproduce**
1. Add the Toast component to your app and add a custom success component based on the `BaseToast` component.
2. Add a custom style to make the width to 100%
3. Open the toast message
4. The message is very small and not readable (see screenshot)

**Expected behavior**
The message should be use the whole size of the screen without any free space left or right. 

**Screenshots**
![Greenshot 2021-11-25 15 21 46](https://user-images.githubusercontent.com/28669616/143458698-52d69e2b-82d5-4c25-9b0e-4d7e3e0987e7.png)

**Code sample**
```
<Toast
  config={{
    success: (props) => (<BaseToast {...props} style={{ width: '100%' }} />)
  }}
/>
```

The sizing only works when using explicit width like `width: 200`.
![Greenshot 2021-11-25 15 38 56](https://user-images.githubusercontent.com/28669616/143460806-1f0843ff-19c3-442a-b817-25b7ee438579.png)

**Environment (please complete the following information):**
 - OS: iOS
 - react-native-toast-message version: v2.0.2
 - react-native version v0.66.3

**Comment by calintamas** [comment]
Hi @sdumjahn, try using the [Dimensions API](https://reactnative.dev/docs/dimensions)

**Comment by jstheoriginal** [comment]
yeah this one threw me for a loop when upgrading. It's probably worth mentioning in the docs that flexbox doesn't work for automatically setting width/height dimensions.

**Comment by sdumjahn** [comment]
@calintamas: Thanks for this advice. I found this workaround a few minutes after I created the thread, but I think that this is a bad solution and I would preferrer to have a solution based on flexbox.

When this is not possible, I think it should be good to mention this workaround in the docs.

**Comment by calintamas** [comment]
I'll have to check though, maybe it could be supported. I think it's the [container style](https://github.com/calintamas/react-native-toast-message/blob/main/src/components/AnimatedContainer.tsx#L127) which breaks it

**Comment by damian-balas** [comment]
> I'll have to check though, maybe it could be supported. I think it's the [container style](https://github.com/calintamas/react-native-toast-message/blob/main/src/components/AnimatedContainer.tsx#L127) which breaks it

Yes, you probably shouldn't center the content. I want to show it on the right side. 
I Had to do a margin: 20% and max width 80% trick. 
That's OK for now... but I'd like to have the possibility to change the root container styles anyway :)

**Comment by calintamas** [comment]
Fixed in `v2.1.0`. 

It should be possible to have a full width Toast via:
```js
<BaseToast style={{ width: '100%' }} />
```

Or change its alignment:

```js
<BaseToast style={{ alignSelf: 'flex-end' }} />
```

**Comment by damian-balas** [comment]
> Fixed in `v2.1.0`.
> 
> It should be possible to have a full width Toast via:
> 
> ```js
> <BaseToast style={{ width: '100%' }} />
> ```
> 
> Or change its alignment:
> 
> ```js
> <BaseToast style={{ alignSelf: 'flex-end' }} />
> ```

@calintamas Sadly only the child element is 80% and aligned to the right.
The issue is that I have a button behind the toast that I want to click.

![image](https://user-images.githubusercontent.com/39343843/146530381-1001b492-649e-4108-bf14-a8d64e0605bc.png)

I cannot click the back button. It would be perfect if I could change the styles of the parent as I mentioned in my previous comment :) Thanks!

**Comment by calintamas** [comment]
@damian-balas Hi, I'll have to think more deeply about it. 

Ideally, the container styles are provided by the lib and there would be no need to change them. I want to avoid possible layout issues occuring due to not having the complete mental model of how it works and making wrong assumptions when changing the styles. 

The main objective is to keep a nice DX in general, while also trying to support as many use cases as I can.

**Comment by calintamas** [comment]
> The issue is that I have a button behind the toast that I want to click.
> ![image](https://user-images.githubusercontent.com/39343843/146530381-1001b492-649e-4108-bf14-a8d64e0605bc.png)

@damian-balas Fixed in [v2.1.1](https://github.com/calintamas/react-native-toast-message/releases/tag/v2.1.1), taps behind the Toast should be possible now :)
